service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/C21OrderEmailContentProvider.java

```diff
@@ -43,24 +48,34 @@ protected C21OrderEmailContentProvider(@Value("${ccd.ui.base.url}")String uiBase
         // Validation within our frontend ensures that the following data is present
         CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
         final String subjectLine = buildSubjectLine(caseData);
-        return ImmutableMap.of(
+        return Map.of(
             "subjectLine", subjectLine,
             "localAuthorityOrCafcass", localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthorityCode),
             "hearingDetailsCallout", buildSubjectLineWithHearingBookingDateSuffix(subjectLine, caseData),
+            "linkToDocStore", mostRecentUploadedC21DocumentUrl(caseData.getC21Orders()),
```

Comment:
I don't like this implementation to be honest. The event should simply send the C21 Order which was attached, we shouldn't reach to the case and find out which one was added. 

Probably logic will be the same but in different place - event itself should be descriptive and know what happened.

Reply:
Why linkToDocStore and not simply linkToDocument? Just curious not 100% sure which one is better.

Reply:
good suggestion on moving into the event itself in place of reaching into the _CaseData_. With regards the _linkToDocStore_, is it ok if I update the template on Gov dot notify and make this _linkToDocument_ as its more readable? @akomisarek